sklearn/datasets/_openml.py

```diff
@@ -81,15 +88,15 @@ def _open_openml_url(openml_path, data_home):
     result : stream
         A stream to the OpenML resource
     """
-    def is_gzip(_fsrc):
+    def is_gzip_encoded(_fsrc):
```

Comment:
Renamed because I found the name confusing. It's not actually a gzipped file or not, but rather that `Content-Encoding == 'gzip'`in the header. So we need to use gzip for local files when this function is False.